### ISSUE TITLE
conditional validation regarding checkboxes

### ISSUE THREAD

**Issue by ambicoder** [thedersen/backbone.validation#284]
I am having trouble implementing conditional validation...in this http://jsbin.com/qudaro/16/ the validation takes place for the input boxes(if these are empty)...in the jsbin you will also see some check boxes...I would like the validation not to take place if these are ticked(checked)...I have looked at the docs regarding conditional validation...nonetheless I cannot get it right.

**Comment by platinumazure** [comment]
You're not actually using the conditional validation at the moment.

A couple of things will make this easier:

  1. You should consider adding a model attribute to represent being closed; in that case, you could set `required` like so:

```javascript
validation: {
    from_monday: {
        required: function (val, attr, computed) {
            return !computed.closed_monday;   // where closed_monday is the checkbox state
        },
        /* etc. */
    }
}
```

2. There will be a lot of repeated logic validating each day. You would probably do better to use a `Backbone.Collection` for each day. The Model which that collection represents can then implement validation only once, and it would apply to each model instance in the collection. You can then use `{ validate: true }` when doing collection set, or however you want to implement the collection update.

3. It might be worthwhile to consider creating an item view (one per day) and instantiate one for each model in the collection (above). That way you can prevent view logic from being duplicated if validation issues arise.

**Comment by ambicoder** [comment]
I want to focus first on 1.How am I going to achieve the representation as being close....for the model attribute related to the check box.

**Comment by platinumazure** [comment]
In your JSBin, you would need to augment the `submit` function to set the attributes according to the checkbox values (add the appropriate IDs to the checkboxes as well):

    input.set({
        from_monday:this.$('#from_mond').val(),
        till_monday:this.$('#till_mond').val(),
        closed_monday: this.$("#closed_mond").prop("checked"),
        from_tuesd:this.$('#from_tuesd').val(), 
        till_tuesd:this.$('#till_tuesd').val(),
        closed_tuesday: this.$("#closed_tuesd").prop("checked")
    }, {validate: true});
    
Then your model will have those attributes.

I also highly recommend using a library like **Backbone.Stickit** or **Backbone.ModelBinder** to quickly and easily bind between view elements and model attributes-- manually setting each time can be a pain when you're working with more complex requirements.

**Comment by ambicoder** [comment]
I have made changes to the code...still though it does not work....the changes are in lines 9 and 53...all of which are related to what it is mentioned here,I have removed Tuesday for reasons of simplicity.

**Comment by ambicoder** [comment]
I have made some progress...probably it is OK now...I will not close the issue yet...I want to be certain,

**Comment by ambicoder** [comment]
I have not completed the backbone part of the code(repeated logic)but for the reason I opened this issue things are OK now...conditional validation works as I want....so I am closing it.
If I need more specific help about it I will open another issue...but for now I am OK
